core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -161,16 +161,36 @@ private[sql] final case class ColumnTableScan(
         statsFor(a).lowerBound <= l && l <= statsFor(a).upperBound
       case EqualTo(l: DynamicReplacableConstant, a: AttributeReference) =>
         statsFor(a).lowerBound <= l && l <= statsFor(a).upperBound
+      case EqualTo(a: AttributeReference, l: Literal) =>
```

Comment:
Sumedh, can you please clarify this for me. We have added extra filter conditions. If they were absent before, although skipping would have been bad, but all rows should have got reflected, isn't it?

Reply:
These conditions were added on code inspection and debugging. The test there was applying the filter due to the DynamicReplaceableConstant condition already present. As mentioned the real issue the numColumnsInStatBlob change below.

Where I got mislead while debugging was that turning off tokenization caused the test to pass causing me to chase the tokenization code while the real problem was that filter was getting skipped for EqualTo when tokenization is disabled.